util/types/datum.go

```diff
@@ -1044,7 +1044,17 @@ func (d *Datum) convertToMysqlYear(sc *variable.StatementContext, target *FieldT
 }
 
 func (d *Datum) convertToMysqlBit(sc *variable.StatementContext, target *FieldType) (Datum, error) {
-	x, err := d.convertToUint(sc, target)
+	var (
```

Comment:
Newly added tests are fundamentally related to changes within this function. Parser will pass a string type `Datum`  to this function when user input `'\0'` or `'100'`, and `getDefaultValue` will pass a string presentation of `bit` to it in the end.  As mentioned in commit message, MySQL treat string as binary string when convert it to `bit`, but TiDB treat it as number literal. This difference cause #2858 and #2717. So new tests cover the string type case and existing tests cover others cases.